runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java

```diff
@@ -1220,10 +1221,13 @@ private KafkaOffsetType asMaximumOffset(
         }
 
         private int nextPartitionData(
+            KafkaKeyFW hashKey,
             KafkaKeyFW key)
         {
             final int partitionCount = leadersByPartitionId.size();
-            final int keyHash = key.length() != -1 ? defaultKeyHash(key) : nextNullKeyHashData++;
+            int keyHash = (hashKey.length() != -1) ? defaultKeyHash(hashKey) :
+                (key.length() != -1) ? defaultKeyHash(key) :
+                    nextNullKeyHashData++;
```

Comment:
`final` can be kept as we don't mutate `keyHash`.
Parentheses around `...length() != -1` can be removed (x2), agree?